pkg/metrics/wal/wal.go

```diff
@@ -626,6 +630,50 @@ func (a *appender) Append(ref storage.SeriesRef, l labels.Labels, t int64, v flo
 	return storage.SeriesRef(series.ref), nil
 }
 
+func (a *appender) AppendHistogram(ref storage.SeriesRef, l labels.Labels, t int64, h *histogram.Histogram) (storage.SeriesRef, error) {
+	series := a.w.series.getByID(chunks.HeadSeriesRef(ref))
```

Comment:
nit: this naming strikes me as odd, because the plural and singular are the same. Thoughts on naming to singleSeries?